source/common/json/json_loader.cc

```diff
@@ -1,125 +1,118 @@
 #include "json_loader.h"
 
-// Do not let jansson leak outside of this file.
+// Do not let RapidJson leak outside of this file.
 #pragma GCC diagnostic push
 #pragma GCC diagnostic ignored "-Wold-style-cast"
-#include "jansson.h"
+#include <rapidjson/error/en.h>
+#include <rapidjson/filereadstream.h>
 #pragma GCC diagnostic pop
 
 namespace Json {
 
 FileLoader::FileLoader(const std::string& file_path) {
-  json_error_t error;
-  json_ = json_load_file(file_path.c_str(), 0, &error);
-  if (!json_) {
-    throw Exception(fmt::format("json parsing error: {} line: {} column: {}", error.text,
-                                error.line, error.column));
+  char buffer[4096];
+
+  FILE* fp = fopen(file_path.c_str(), "r");
+  if (!fp) {
+    throw Exception(fmt::format("File:{} doesn't exist.", file_path));
+  }
+
+  rapidjson::FileReadStream fs(fp, buffer, sizeof(buffer));
```

Comment:
what is buffer used for here? Why does it have to be specified? Why 4K? Seems odd there is no API to just deal with buffering internally.